native/src/main/java/io/ballerina/stdlib/grpc/builder/syntaxtree/SyntaxTreeGenerator.java

```diff
@@ -111,14 +109,30 @@ public static SyntaxTree generateSyntaxTree(StubFile stubFile, boolean isRoot) {
                     "ballerina", "grpc"
             );
             imports = imports.add(importForGrpc);
-            if (checkForImportsInStub(stubFile, GOOGLE_PROTOBUF_TIMESTAMP_PROTO)
-                    || checkForImportsInStub(stubFile, GOOGLE_PROTOBUF_DURATION_PROTO)) {
-                ImportDeclarationNode importForTime = Imports.getImportDeclarationNode(
-                        "ballerina", "time"
-                );
-                imports = imports.add(importForTime);
+        }
+        boolean importTime = false;
+        for (ServiceStub serviceStub : stubFile.getStubList()) {
+            List<Method> methodList = new ArrayList<>();
+            methodList.addAll(serviceStub.getUnaryFunctions());
+            methodList.addAll(serviceStub.getClientStreamingFunctions());
+            methodList.addAll(serviceStub.getServerStreamingFunctions());
+            methodList.addAll(serviceStub.getBidiStreamingFunctions());
+
+            if (checkForImportsInServices(methodList, "time:Utc")
+                    || checkForImportsInServices(methodList, "time:Seconds")) {
+                importTime = true;
             }
         }
+        if (checkForImportsInMessageMap(stubFile, "Timestamp") ||
+                checkForImportsInMessageMap(stubFile, "Duration")) {
+            importTime = true;
+        }
+        if (importTime) {
+            ImportDeclarationNode importForTime = Imports.getImportDeclarationNode(
+                    "ballerina", "time"
+            );
+            imports = imports.add(importForTime);
```

Comment:
@MadhukaHarith92 is this change adding an import to the correct generated file. there can be scenarios where we need to add the import to both the root file and dependent file.

Reply:
Yes. Added a test case where the import gets added to both root and dependent files in 7db7f64